aws/resource_aws_media_package_channel.go

```diff
@@ -0,0 +1,156 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/mediapackage"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsMediaPackageChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMediaPackageChannelCreate,
+		Read:   resourceAwsMediaPackageChannelRead,
+		Update: resourceAwsMediaPackageChannelUpdate,
+		Delete: resourceAwsMediaPackageChannelDelete,
+		Importer: &schema.ResourceImporter{
```

Comment:
Acceptance testing and documentation are missing for import:

```go
func TestAccAWSMediaPackageChannel_basic(t *testing.T) {
	resourceName := "aws_media_package_channel.test"

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAwsMediaPackageChannelDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccMediaPackageChannelConfig(acctest.RandString(5)),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAwsMediaPackageChannelExists(resourceName),
					resource.TestMatchResourceAttr(resourceName, "ingest_endpoints.0.password", regexp.MustCompile("^[0-9a-f]*$")),
					resource.TestMatchResourceAttr(resourceName, "ingest_endpoints.0.url", regexp.MustCompile("^https://")),
					resource.TestMatchResourceAttr(resourceName, "ingest_endpoints.0.username", regexp.MustCompile("^[0-9a-f]*$")),
					resource.TestMatchResourceAttr(resourceName, "ingest_endpoints.1.password", regexp.MustCompile("^[0-9a-f]*$")),
					resource.TestMatchResourceAttr(resourceName, "ingest_endpoints.1.url", regexp.MustCompile("^https://")),
					resource.TestMatchResourceAttr(resourceName, "ingest_endpoints.1.username", regexp.MustCompile("^[0-9a-f]*$")),
				),
			},
			{
				ResourceName: resourceName,
				ImportState: true,
				ImportStateVerify: true,
			},
		},
	})
}
```

``````markdown
## Import

Media Package Channels can be imported via the channel ID, e.g.

```
$ terraform import aws_media_package_channel.kittens kittens-channel
```
``````